File: shared/FlixorKit/Sources/FlixorKit/Services/EditionService.swift
//
//  EditionService.swift
//  FlixorKit
//
//  Edition title extraction (top-level or Media[0])
//

import Foundation

public enum EditionService {

    /// Prefer top-level editionTitle, then first media's editionTitle.
    public static func extractEditionTitle(
        topLevelEdition: String?,
        firstMediaEdition: String?
    ) -> String? {
        if let edition = topLevelEdition, !edition.isEmpty {
            return edition
        }
        if let edition = firstMediaEdition, !edition.isEmpty {
            return edition
        }
        return nil
    }

    public static func extractEditionTitle(
        topLevelEdition: String?,
        media: [PlexMedia]?
    ) -> String? {
        return extractEditionTitle(
            topLevelEdition: topLevelEdition,
            firstMediaEdition: media?.first?.editionTitle
        )
    }
}

File: packages/core/src/services/EditionService.ts
/**
 * Edition title extraction (top-level or Media[0]).
 * Matches FlixorKit EditionService logic.
 */

export class EditionService {
  static extractEditionTitle(
    topLevelEdition: string | undefined,
    media: { editionTitle?: string }[] | undefined
  ): string | undefined {
    if (topLevelEdition?.trim()) {
      return topLevelEdition;
    }
    const firstMediaEdition = media?.[0]?.editionTitle;
    if (firstMediaEdition?.trim()) {
      return firstMediaEdition;
    }
    return undefined;
  }
}
